app/workers/pull_request_monitor_handlers/path_based_labeler.rb

```diff
@@ -0,0 +1,27 @@
+class PullRequestMonitorHandlers::PathBasedLabeler
+  include Sidekiq::Worker
+  sidekiq_options :queue => :miq_bot
+
+  include BranchWorkerMixin
+
+  def perform(branch_id)
+    return unless find_branch(branch_id, :pr)
+    return unless verify_branch_enabled
+
+    process_branch
+  end
+
+  private
+
+  def process_branch
+    label_rules.each do |rule|
+      if diff_file_names.any? { |file_name| file_name =~ Regexp.new(rule.pattern) }
+        GithubService.add_labels_to_an_issue(fq_repo_name, pr_number, [rule.label])
```

Comment:
I believe that logic is just moved in to GithubService::Issue now:

https://github.com/manageiq/miq_bot/blob/637e49616839c5765d9c2f050c43338afc396f04/lib/github_notification_monitor.rb#L132

and

https://github.com/manageiq/miq_bot/blob/637e49616839c5765d9c2f050c43338afc396f04/lib/github_service/issue.rb#L21-L23

The difference for our case here is, unfortunately, that we're not dealing with the already-fetched issue response from Github but processing a branch. 

Will have to ponder further 😕